File: benchmarks/python/guessing_game.py
import random


def main() -> None:
    secret = random.randint(1, 10)
    tries = 0
    print("I picked a number from 1 to 10.")
    while True:
        try:
            text = input("your guess: ")
        except EOFError:
            return
        try:
            guess = int(text)
        except ValueError:
            print("numbers only, please")
            continue
        tries += 1
        if guess == secret:
            print(f"You got it in {tries} tries!")
            return
        if guess < secret:
            print("higher")
        else:
            print("lower")


if __name__ == "__main__":
    main()

File: benchmarks/rust/guessing_game.rs
use std::io::{self, Write};
use std::time::{SystemTime, UNIX_EPOCH};

fn ask(prompt: &str) -> Option<String> {
    print!("{}", prompt);
    io::stdout().flush().unwrap();
    let mut line = String::new();
    let read = io::stdin().read_line(&mut line).ok()?;
    if read == 0 {
        return None;
    }
    Some(line.trim_end().to_string())
}

fn main() {
    let nanos = SystemTime::now()
        .duration_since(UNIX_EPOCH)
        .unwrap()
        .subsec_nanos();
    let secret = (nanos % 10 + 1) as i64;
    let mut tries = 0;
    println!("I picked a number from 1 to 10.");
    loop {
        let Some(text) = ask("your guess: ") else {
            return;
        };
        let Ok(guess) = text.parse::<i64>() else {
            println!("numbers only, please");
            continue;
        };
        tries += 1;
        if guess == secret {
            println!("You got it in {} tries!", tries);
            return;
        } else if guess < secret {
            println!("higher");
        } else {
            println!("lower");
        }
    }
}
